nym-data-observatory/src/chain_scraper/webhook.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2025 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use crate::config::data_observatory::{HttpAuthenticationOptions, WebhookConfig};
+use crate::models::WebhookPayload;
+use anyhow::Context;
+use async_trait::async_trait;
+use nym_validator_client::nyxd::{Any, Msg, MsgSend, Name};
+use nyxd_scraper_psql::{
+    MsgModule, NyxdScraperTransaction, ParsedTransactionResponse, ScraperError,
+};
+use reqwest::{Client, Url};
+use tracing::{error, info};
+use utoipa::r#gen::serde_json;
+
+pub struct WebhookModule {
+    webhooks: Vec<Webhook>,
+}
+
+impl WebhookModule {
+    pub fn new(config: crate::config::Config) -> anyhow::Result<Self> {
+        let webhooks = config
+            .data_observatory_config
+            .webhooks
+            .iter()
+            .map(|watcher_cfg| Webhook::new(watcher_cfg.clone()))
+            .collect::<anyhow::Result<Vec<_>>>()?;
+        Ok(Self { webhooks })
+    }
+}
+
+#[async_trait]
+impl MsgModule for WebhookModule {
+    fn type_url(&self) -> String {
+        <MsgSend as Msg>::Proto::type_url()
+    }
+
+    async fn handle_msg(
+        &mut self,
+        index: usize,
+        _msg: &Any,
+        tx: &ParsedTransactionResponse,
+        _storage_tx: &mut dyn NyxdScraperTransaction,
+    ) -> Result<(), ScraperError> {
+        let message = serde_json::to_value(tx.parsed_messages.get(&index)).ok();
+
+        let payload = WebhookPayload {
+            height: tx.height.value(),
+            message_index: index as u64,
+            transaction_hash: tx.hash.to_string(),
+            message,
+        };
+
+        println!(
+            "->>>>>>>>>>>>>>>>>>>>>>>>> {}",
+            serde_json::to_string(&payload).unwrap()
+        );
+
+        for webhook in self.webhooks.clone() {
+            let payload = payload.clone();
+            tokio::spawn(async move {
```

Comment:
rather than spawning every request as we get them, would it maybe make sense to have a `TaskTracker` or `JoinSet` on `WebhookModule` itself to keep track of all of those?
perhaps with some kind of queue/limit so we wouldn't suddenly spawn 1000 webhooks and starve everything else